File: app/src/main/java/com/taipei/yanghaobo/kunu/ui/edit/DogDetailEditVMFactory.java
package com.taipei.yanghaobo.kunu.ui.edit;

import android.arch.lifecycle.ViewModel;
import android.arch.lifecycle.ViewModelProvider;
import android.support.annotation.NonNull;

import com.taipei.yanghaobo.kunu.data.KunuRepository;

public class DogDetailEditVMFactory extends ViewModelProvider.NewInstanceFactory {

    private final KunuRepository mKunuRepository;
    private final int mId;

    public DogDetailEditVMFactory(KunuRepository kunuRepository, int id) {
        this.mKunuRepository = kunuRepository;
        this.mId = id;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) (new DogDetailEditViewModel(mKunuRepository, mId));
    }
}

File: app/src/main/java/com/taipei/yanghaobo/kunu/ui/edit/DogDetailEditVMFactory.kt
package com.taipei.yanghaobo.kunu.ui.edit

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider

import com.taipei.yanghaobo.kunu.data.KunuRepository

class DogDetailEditVMFactory(private val mKunuRepository: KunuRepository, private val mId: Int) :
  ViewModelProvider.NewInstanceFactory() {

  override fun <T : ViewModel> create(modelClass: Class<T>): T {
    return DogDetailEditViewModel(mKunuRepository, mId) as T
  }
}
